src/gui/mzroll/eicwidget.cpp

```diff
@@ -873,6 +873,15 @@ void EicWidget::clearPlot() {
 	scene()->setSceneRect(10, 10, this->width() - 10, this->height() - 10);
 }
 
+void EicWidget::unSetPeakTableGroup(PeakGroup* group)
+{
+    // Peak table is being deleted. Making sure the selected group is not holding any garbage value;
+    if(eicParameters->selectedGroup ==  group) {
+        eicParameters->selectedGroup = nullptr;
+		setMzRtWindow(0, 0, 0, 0);
```

Comment:
@saifulbkhan setMzRtWindow is the same as setMzSlice. Let's remove this function, create a slice with (0,0,0,0) parameters and call setMzSlice here.

Reply:
Alright. Will do.